### PR TITLE
Skip populate function proposal in Experimental_ArrayMinItems

### PR DESCRIPTION
**Pull request by MarekBodingerBA**
### Reasons for making this change

As we use our custom implementation of file upload, we would like to have case-by-case method how to control population of arrays in `Experimental_ArrayMinItems` (and there might be need from others with advanced usages). 

Our files implementation yields this schema:
https://github.com/bratislava/konto.bratislava.sk/blob/8ce56b7fa4a62c7dd67351b617ba5ad13bc363b8/next/schema-generator/generator/functions.ts#L372

We would like to use `{ populate: 'requiredOnly' }`. It works fine for our [multiselects](https://github.com/bratislava/konto.bratislava.sk/blob/8ce56b7fa4a62c7dd67351b617ba5ad13bc363b8/next/schema-generator/generator/functions.ts#L113) and [checkbox group](https://github.com/bratislava/konto.bratislava.sk/blob/8ce56b7fa4a62c7dd67351b617ba5ad13bc363b8/next/schema-generator/generator/functions.ts#L326), as they are handled by the [isMultiSelect condition](https://github.com/rjsf-team/react-jsonschema-form/blob/af8932a60fb77250dd238ea32257731d3d8308f9/packages/utils/src/schema/getDefaultFormState.ts#L401), but it populates our files array with `[null]`. Therefore I would like to implement `skipPopulate` (name is up for consideration) function that would allow us to have a custom logic to skip population similar to what `isMultiSelect` does.

As a temporary solution, we have our own [fork](https://github.com/bratislava/react-jsonschema-form/commit/df33db5638ce3626a72f8b8411df498a8a8ae5fc) that allows us to add `overrideArrayMinItemsBehaviour` to schema, but this might be better and more scalable solution.

What do you think? Should I continue with this idea?

### Checklist

- [x] **I'm updating documentation**
  - [x] I've [checked the rendering](https://rjsf-team.github.io/react-jsonschema-form/docs/contributing) of the Markdown text I've added
- [x] **I'm adding or updating code**
  - [x] I've added and/or updated tests. I've run `npm run test:update` to update snapshots, if needed.
  - [x] I've updated [docs](https://rjsf-team.github.io/react-jsonschema-form/docs) if needed
  - [x] I've updated the [changelog](https://github.com/rjsf-team/react-jsonschema-form/blob/main/CHANGELOG.md) with a description of the PR
- [x] **I'm adding a new feature**
  - [ ] I've updated the playground with an example use of the feature

### CODE REVIEW

**Comment by heath-freenome** [comment]
@MarekBodingerBA I like the concept. The name `skipPopulate` might be confusing, especially as someone could think it is a boolean. How does the name `computeSkipPopulate` sound? Please make sure to update documentation and provide an example in the playground. And maintain 100% test coverage for these changes. And finally, update the `CHANGELOG.md` under the `5.18.0` section. If you can get this in within the next week we can release right afterwards

**Comment by MarekBodingerBA** [comment]
Perfect, that all makes sense!

However, how should I proceed with a playground example? People should provide their own code. I am not sure I should add code input to the playground.

**Comment by heath-freenome** [comment]
> Perfect, that all makes sense!
> 
> However, how should I proceed with a playground example? People should provide their own code. I am not sure I should add code input to the playground.

Great question! Thinking about it more, this isn't a toggle on situation like the rest of the experimental options. So maybe a playground example just doesn't make sense.

**Comment by MarekBodingerBA** [comment]
Implemented as you requested, feel free to give a feedback.

File: packages/docs/docs/api-reference/form-props.md [hunk 1]
```diff
@@ -83,6 +83,9 @@ Optional enumerated flag controlling how array minItems are populated, defaultin
 | `requiredOnly` | Ignore `minItems` on a field when calculating defaults unless the field is required.                                                               |
 | `never`        | Ignore `minItems` on a field when calculating defaults for required and non-required. Value will set only if defined `default` and from `formData` |
 
+#### `arrayMinItems.computeSkipPopulate`
+A function that determines whether to skip populating the array with default values based on the provided validator, schema, and root schema. If the function returns `true`, the array will not be populated with default values. If the function returns `false`, the array will be populated with default values according to the `populate` option.
```
Comment (heath-freenome) [inline packages/docs/docs/api-reference/form-props.md:87]: I think providing the expected signature of this method and perhaps a minimal `Typescript` example would be helpful?

Comment (MarekBodingerBA) [reply packages/docs/docs/api-reference/form-props.md:88]: I added an example.

File: packages/utils/src/types.ts [hunk 2]
```diff
@@ -41,6 +41,20 @@ export type Experimental_ArrayMinItems = {
    * - `never`: Ignore `minItems` on a field even the field is required.
    */
   populate?: 'all' | 'requiredOnly' | 'never';
+  /** A function that determines whether to skip populating the array with default values based on the provided validator,
+   * schema, and root schema.
+   * If the function returns true, the array will not be populated with default values.
+   * If the function returns false, the array will be populated with default values according to the `populate` option.
+   * @param validator - An implementation of the `ValidatorType` interface that is used to detect valid schema conditions
+   * @param schema - The schema for which resolving a condition is desired
+   * @param rootSchema - The root schema that will be forwarded to all the APIs
```
Comment (heath-freenome) [inline packages/utils/src/types.ts:50]:
```suggestion
   * @param [rootSchema] - The root schema that will be forwarded to all the APIs
```

File: packages/docs/docs/api-reference/form-props.md [hunk 3]
```diff
@@ -84,6 +84,51 @@ Optional enumerated flag controlling how array minItems are populated, defaultin
 | `requiredOnly` | Ignore `minItems` on a field when calculating defaults unless the field is required.                                                               |
 | `never`        | Ignore `minItems` on a field when calculating defaults for required and non-required. Value will set only if defined `default` and from `formData` |
 
+#### `arrayMinItems.computeSkipPopulate`
+
+A function that determines whether to skip populating the array with default values based on the provided validator, schema, and root schema. If the function returns `true`, the array will not be populated with default values. If the function returns `false`, the array will be populated with default values according to the `populate` option.
```
Comment (heath-freenome) [inline packages/docs/docs/api-reference/form-props.md:89]:
```suggestion
The signature for this property is as follow:

##### computeSkipPopulate <T = any, S extends StrictRJSFSchema = RJSFSchema, F extends FormContextType = any>()

A function that determines whether to skip populating the array with default values based on the provided validator, schema, and root schema.
 If the function returns `true`, the array will not be populated with default values.
 If the function returns `false`, the array will be populated with default values according to the `populate` option.

###### Parameters

- validator: ValidatorType<T, S, F> - An implementation of the `ValidatorType` interface that is used to detect valid schema conditions
- schema: S - The schema for which resolving a condition is desired
- [rootSchema]: S - The root schema that will be forwarded to all the APIs

###### Returns

- boolean: A boolean indicating whether to skip populating the array with default values.

```

File: packages/docs/docs/api-reference/form-props.md [hunk 4]
```diff
@@ -84,6 +84,51 @@ Optional enumerated flag controlling how array minItems are populated, defaultin
...
+
```
Comment (heath-freenome) [inline packages/docs/docs/api-reference/form-props.md:90]:
```suggestion

##### Example

```

**Comment by heath-freenome** [comment]
@MarekBodinger Great work... I've added a few suggetions for documentation changes. Also, can you rebase and resolve conflicts

**Comment by MarekBodingerBA** [comment]
> @MarekBodinger Great work... I've added a few suggetions for documentation changes. Also, can you rebase and resolve conflicts

Thank you, I committed the suggestions and merged main.

**Review by heath-freenome — APPROVED** [review]

**Pull request merged**